expertsystem/data.py

```diff
@@ -352,29 +353,28 @@ def find(self, search_term: Union[int, str]) -> Particle:
             f"Cannot search for a search term of type {type(search_term)}"
         )
 
-    def find_subset(
-        self, search_term: Union[int, str]
+    def filter(  # noqa: A003
+        self, function: Callable[[Particle], bool]
```

Comment:
So its `ParticleCollection.filter(lambda p: p.charge == 1)` vs `[p for p in ParticleCollection if p.charge == 1]`

In my view the filter function is unnecessary, if `ParticleCollection` can be a basic datatype like a `set` (see previous comment).

Reply:
https://github.com/ComPWA/expertsystem/pull/260#discussion_r493306662

Reply:
Also see
https://pwa--260.org.readthedocs.build/projects/expertsystem/en/260/usage/particles.html
(unfortunately no cell output in the PR rendering on RTD, while that's actually the point of the modifications to `ParticleCollection.__repr__` and `Particle.__repr__`.

Reply:
So since `ParticleCollection` is not a primitive type (like `set`), this is totally fine. Because we want to have a ParticleCollection again after the filtering.